spigot/src/main/java/fr/freebuild/playerjoingroup/spigot/MessagesManager.java

```diff
@@ -174,10 +174,10 @@ public void send(byte[] msg) throws IOException {
     }
 
     private void onPlayerLeave(Packet packet) {
-        String playerName = packet.getParams().get("PLAYER_NAME");
-        String message = Utils.getConfigString("QuitMessage");
-        message = Utils.format(message, FormatParam.PLAYER, playerName);
-        getServer().broadcastMessage(message);
+        if (canDisplayMessage(packet, "essentials.silentquit"))  {
+            String message = Utils.getPlayerLeaveMessage(packet.getParams().get("PLAYER_NAME"));
+            getServer().broadcastMessage(message);
+        }
```

Comment:
Doesn't work because the player already leave. Find a way to check permission on offline player or move the check elsewhere